### PR TITLE
(cloud-merge) Support BE smooth upgrade

### PR DESCRIPTION
**Pull request by deardeng**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: be/src/service/http_service.cpp [hunk 1]
```diff
@@ -225,6 +226,10 @@ Status HttpService::start() {
         register_local_handler(engine.to_local());
     }
 
+    // shrink memory for starting co-exist process during upgrade
+    ShrinkMemAction* shrink_mem_action = _pool.add(new ShrinkMemAction());
+    _ev_http_server->register_handler(HttpMethod::GET, "/api/shrink_mem", shrink_mem_action);
```
Comment (gavinchou) [inline be/src/service/http_service.cpp:231]: Put these new lines before line 222, before this section
```
    auto& engine = _env->storage_engine();
    if (config::is_cloud_mode()) {
        register_cloud_handler(engine.to_cloud());
    } else {
        register_local_handler(engine.to_local());
    }
```

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java [hunk 2]
```diff
@@ -449,6 +447,8 @@ private void getCloudBackends() {
             // local - remote > 0, drop bes from local
             checkToDelCluster(remoteClusterIdToPB, localClusterIds, clusterIdToBackend);
 
+            clusterIdToBackend = cloudSystemInfoService.getCloudClusterIdToBackend();
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:450]: Is this a bug fix not related to smooth upgrade?

File: fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java [hunk 3]
```diff
@@ -480,6 +481,9 @@ public static boolean statsTblAvailable() {
             return false;
         }
         if (Config.isCloudMode()) {
+            if (!((CloudSystemInfoService) Env.getCurrentSystemInfo()).availableBackendsExists()) {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java:484]: Is this another bug fix not related to smooth upgrade?

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java (hunk 2 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/cloud/catalog/CloudClusterChecker.java:450]: yes,  not related to smooth upgrade

File: fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java (hunk 3 above)
Comment (deardeng) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/util/StatisticsUtil.java:484]: yes, just discovered during testing

**Comment by deardeng** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**